controllers/rule_controller.go

```diff
@@ -41,13 +48,56 @@ func (r *RuleReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 
 	ctx := context.Background()
 	_ = r.Log.WithValues("rule", req.NamespacedName)
-	rulesList := &oathkeeperv1alpha1.RuleList{}
 
-	err := r.List(ctx, rulesList)
+	var rule oathkeeperv1alpha1.Rule
+	skipValidation := false
+
+	if err := r.Get(ctx, req.NamespacedName, &rule); err != nil {
+		if apierrs.IsNotFound(err) {
+			skipValidation = true
+		} else {
+			return ctrl.Result{}, err
+		}
+	}
+
+	if !skipValidation {
+		if err := rule.ValidateWith(r.ValidationConfig); err != nil {
+			rule.Status.Validation = &oathkeeperv1alpha1.Validation{}
+			rule.Status.Validation.Valid = boolPtr(false)
+			e := err.Error()
+			rule.Status.Validation.Error = &e
+			r.Log.Info("validation error in Rule %s/%s: \"%s\"", rule.Namespace, rule.Name, e)
+			if err := r.Update(ctx, &rule); err != nil {
+				r.Log.Error(err, "unable to update Rule status")
+				return ctrl.Result{}, err
+			}
+			// continue, as validation can't be fixed by requeuing request and we still have to update the configmap
+		} else {
+			// rule valid - set the status
+			rule.Status.Validation = &oathkeeperv1alpha1.Validation{}
+			rule.Status.Validation.Valid = boolPtr(true)
+			if err := r.Update(ctx, &rule); err != nil {
+				r.Log.Error(err, "unable to update Rule status")
+				return ctrl.Result{}, err
```

Comment:
We can tell the controller to requeue by setting field in the result, imo we should do that because otherwise controller will re-reconcile the request with error log but since it passed the validation it should not be reconciled with error.
```suggestion
				return ctrl.Result{Requeue: true}, nil
```

Reply:
Resolved in d8257dc